.ci/test_test_manager.py

```diff
@@ -11,79 +11,41 @@
             warnings.simplefilter('ignore', ResourceWarning)  # pylint: disable=no-member,undefined-variable
 
         self.backend = get_test_backend_name()
-        self.profile_manager = TemporaryProfileManager(backend=self.backend)
+        self.test_manager = TestManager()
 
     def tearDown(self):
-        self.profile_manager.destroy_all()
-
-    def test_create_db_cluster(self):
-        self.profile_manager.create_db_cluster()
-        self.assertTrue(pgtest.is_server_running(self.profile_manager.pg_cluster.cluster))
+        self.test_manager.destroy_all()
 
-    def test_create_aiida_db(self):
-        self.profile_manager.create_db_cluster()
-        self.profile_manager.create_aiida_db()
-        self.assertTrue(self.profile_manager.postgres.db_exists(self.profile_manager.profile_info['database_name']))
-
-    def test_create_use_destroy_profile(self):
+    def test_pgtest_argument(self):
         """
-        Test temporary test profile creation
-
-        * The profile gets created, the dbenv loaded
-        * Data can be stored in the db
-        * reset_db deletes all data added after profile creation
-        * destroy_all removes all traces of the test run
+        Create a temporary profile, passing the pgtest argument.
         """
-        with Capturing() as output:
-            self.profile_manager.create_profile()
-
-        self.assertTrue(self.profile_manager.root_dir_ok, msg=output)
-        self.assertTrue(self.profile_manager.config_dir_ok, msg=output)
-        self.assertTrue(self.profile_manager.repo_ok, msg=output)
-        from aiida.manage.configuration.settings import AIIDA_CONFIG_FOLDER
-        self.assertEqual(AIIDA_CONFIG_FOLDER, self.profile_manager.config_dir, msg=output)
-
-        from aiida.orm import load_node
-        from aiida.plugins import DataFactory
-        data = DataFactory('dict')(dict={'key': 'value'})
-        data.store()
-        data_pk = data.pk
-        self.assertTrue(load_node(data_pk))
-
-        with self.assertRaises(TestManagerError):
-            self.test_create_aiida_db()
+        from aiida.common.files import which
 
-        self.profile_manager.reset_db()
-        with self.assertRaises(Exception):
-            load_node(data_pk)
+        # this should fail
+        pgtest = {'pg_ctl': 'notapath'}
+        with self.assertRaises(AssertionError):
+            self.test_manager.use_temporary_profile(backend=self.backend, pgtest=pgtest)
 
-        temp_dir = self.profile_manager.root_dir
-        self.profile_manager.destroy_all()
-        with self.assertRaises(Exception):
-            self.profile_manager.postgres.db_exists(self.profile_manager.dbinfo['db_name'])
-        self.assertFalse(os.path.exists(temp_dir))
-        self.assertIsNone(self.profile_manager.root_dir)
-        self.assertIsNone(self.profile_manager.pg_cluster)
+        # pg_ctl is what PGTest also looks for (although it might be more clever)
+        pgtest = {'pg_ctl': which('pg_ctl')}
```

Comment:
Maybe we should use the `which` from `pgtest`? And/or add another test that tests what we expect with `which` from AiiDA is the same that `pgtest` expects. There is a tight connection between AiiDA and `pgtest` now, but that might change.

Reply:
> Maybe we should use the which from pgtest? 

Since pgtest is a bit more clever than the standard "which", we could do that.
I'll make the change.

> And/or add another test that tests what we expect with which from AiiDA is the same that pgtest expects.

I don't think that needs to be the case, and it's not what I'm testing here - I'm just trying to see of the `TemporaryProfileManager` works if I give it *some* valid path to `pg_ctl` (could be a different one than the one autodiscovered by pgtest).

Reply:
My main question for you was to check whether you are ok with me removing the `pgtest` argument from the internal functions of the `TemporaryProfileManager` (since I guess you were the main user) - I guess as long as it is still accessible, it is fine with you, right?

Reply:
Yes, absolutely. I was using it from the `test_manager`, which run the constructor I think, so for me this would work nicely.